smslibrary/src/test/java/com/eis/smslibrary/SMSDeliveredBroadcastReceiverTest.java

```diff
@@ -0,0 +1,79 @@
+package com.eis.smslibrary;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+
+import com.eis.smslibrary.listeners.SMSDeliveredListener;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author Marco Cognolato
+ */
+public class SMSDeliveredBroadcastReceiverTest {
+    private SMSDeliveredBroadcastReceiver receiver;
+    private SMSDeliveredListener listenerMock = mock(SMSDeliveredListener.class);
+    private Context contextMock = mock(Context.class);
+    private static final SMSPeer VALID_PEER = new SMSPeer("+393423541601");
+    private static final String VALID_TEXT = "Some random text to send";
+    private static final ArrayList<String> MESSAGE_TO_SEND = new ArrayList<>();
+    private static final SMSMessage VALID_MESSAGE = new SMSMessage(VALID_PEER, VALID_TEXT);
+
+    @Before
+    public void setup() {
+        MESSAGE_TO_SEND.clear();
+        MESSAGE_TO_SEND.add(VALID_TEXT);
+        receiver = spy(new SMSDeliveredBroadcastReceiver(MESSAGE_TO_SEND, listenerMock, VALID_PEER));
+    }
+
+    @Test
+    public void constructor_doesntThrow() {
+        try {
+            new SMSDeliveredBroadcastReceiver(MESSAGE_TO_SEND, listenerMock, VALID_PEER);
+            //Success
+        } catch (Exception e) {
+            fail();
+        }
+    }
```

Comment:
Test fails automatically when an Exception is thrown.
Is this `try-catch` necessary?
It would be appreciated if you let me know...

Reply:
the try-catch is not _necessarily_ required, but it's used to clearly know what I'm testing for